tests/test_core_distribution.cc

```diff
@@ -140,6 +140,48 @@ TYPED_TEST_P(DistributionTest, test_multiply_by_scalar) {
   EXPECT_EQ(actual, scaled);
 };
 
+TYPED_TEST_P(DistributionTest, test_equal) {
+  const Eigen::Index n = 3;
+  const Eigen::VectorXd mean = Eigen::VectorXd::Random(n, 1);
+  const Eigen::VectorXd var = Eigen::VectorXd::Random(n, 1).array().square();
+
+  const MarginalDistribution dist1(mean, var);
+  const MarginalDistribution dist2(mean, var);
+
+  EXPECT_TRUE(dist1 == dist2);
+};
+
+TYPED_TEST_P(DistributionTest, test_approximately_equal) {
+  const Eigen::Index n = 4;
+  const Eigen::VectorXd mean = Eigen::VectorXd::Random(n, 1);
+  const Eigen::VectorXd var = Eigen::VectorXd::Random(n, 1).array().square();
+
+  {
+    const MarginalDistribution dist1(mean, var);
+    const MarginalDistribution dist2(mean, var);
+    EXPECT_TRUE(dist1.approximately_equal(
+        dist2, std::numeric_limits<double>::epsilon()));
+  }
+
+  {
+    const MarginalDistribution dist1(mean, var);
+    const MarginalDistribution dist2(mean + Eigen::VectorXd::Constant(n, 1e-4),
+                                     var);
+    EXPECT_FALSE(dist1.approximately_equal(
+        dist2, std::numeric_limits<double>::epsilon()));
+    EXPECT_TRUE(dist1.approximately_equal(dist2, 1e-3));
+  }
+
+  {
+    const MarginalDistribution dist1(mean, var);
+    const MarginalDistribution dist2(mean + Eigen::VectorXd::Constant(n, 1e-5),
+                                     var);
+    EXPECT_FALSE(dist1.approximately_equal(
+        dist2, std::numeric_limits<double>::epsilon()));
+    EXPECT_FALSE(dist1.approximately_equal(dist2, 1e-6));
+  }
```

Comment:
The `DistributionTest` case is a helper to enable the testing of all possible types of distributions (`MarginalDistribution` and `JointDistribution`).

https://github.com/swift-nav/albatross/blob/master/tests/test_core_distribution.cc#L208

To use the `DistributionTest` fixture you create a distribution [like this](https://github.com/swift-nav/albatross/blob/master/tests/test_core_distribution.cc#L145C1-L146C40
):
```
  TypeParam test_case;
  const auto dist = test_case.create();
```
and then the test you write will get executed on all current and future types. So you'll probably want to do something like this:

```suggestion
  TypeParam test_case;
  const auto dist = test_case.create();
  {
    auto copied = dist;
    EXPECT_TRUE(dist.approximately_equal(
        copied, std::numeric_limits<double>::epsilon()));
  }

  {
    auto perturbed = dist;
    purturbed.mean += Eigen::VectorXd::Constant(n, 1e-4);
    EXPECT_FALSE(dist.approximately_equal(
        purturbed, std::numeric_limits<double>::epsilon()));
    EXPECT_TRUE(dist.approximately_equal(purturbed, 1e-3));
  }
```
Note that for that to pass you'll need to have `approximately_equal` work for both `MarginalDistribution` and `JointDistribution`, which is probably a good idea anyway since these distribution classes share the same base class and most other methods ... but could arguably be out of scope.

An alternative if you don't want to add the corresponding method to `JointDistribution` is to remove the `DistributionTest` and just do,
```
TEST(test_core_distribution, test_marginal_distribution_approximately_equal) {
```

Reply:
I decided to add the `approximately_equal` for both.